ResetPasswordBundle/src/tests/UnitTests/Persistence/PWResetRequestRepositoryTraitTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SymfonyCasts\Bundle\ResetPassword\tests\UnitTests\Persistence;
+
+use SymfonyCasts\Bundle\ResetPassword\Persistence\PasswordResetRequestRepositoryInterface;
+use SymfonyCasts\Bundle\ResetPassword\tests\Fixtures\PWResetRequestRepositoryTraitTestFixture;
+use PHPUnit\Framework\TestCase;
+use SymfonyCasts\Bundle\ResetPassword\Persistence\Repository\PasswordResetRequestRepositoryTrait;
+
+class PWResetRequestRepositoryTraitTest extends TestCase
+{
+    public function methodDataProvider(): \Generator
+    {
+        yield ['persistPasswordResetRequest'];
+        yield ['findPasswordResetRequest'];
+        yield ['getMostRecentNonExpiredRequestDate'];
+        yield ['removeResetRequest'];
+    }
+
+    /**
+     * @test
+     * @dataProvider methodDataProvider
+     */
+    public function hasMethod(string $method): void
+    {
+        self::assertTrue(method_exists(PasswordResetRequestRepositoryTrait::class, $method));
+    }
+
+    /** @test */
+    public function traitIsCompatibleWithInterface(): void
```

Comment:
Hmm, interesting. So a bullet-proof way of doing this would be to:

A) Embed a "fake" class inside this file that implements the interface and uses the trait.
B) Try to instantiate that object in a test method and... just assert that it implements the interface

If the trait doesn't implement the interface correctly, the test method in (B) would explode even before the assertion.

Reply:
done